hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/HeapData.java

```diff
@@ -93,8 +93,7 @@ public int getType() {
     @Override
     public int getHeapCost() {
         // reference (assuming compressed oops)
-        int objectRef = INT_SIZE_IN_BYTES;
-        return objectRef + (payload != null ? ARRAY_HEADER_SIZE_IN_BYTES + payload.length : 0);
+        return REFERENCE_COST_IN_BYTES + (payload != null ? ARRAY_HEADER_SIZE_IN_BYTES + payload.length : 0);
```

Comment:
Currently, `ARRAY_HEADER_SIZE_IN_BYTES ` calculation is out of scope of this PR. But in a future PR, i am planning to address it.